x-pack/legacy/plugins/monitoring/index.ts

```diff
@@ -0,0 +1,131 @@
+  'monitoring.ui.container.logstash.enabled',
+  'monitoring.tests.cloud_detector.enabled',
+  'monitoring.kibana.collection.interval',
+  'monitoring.ui.elasticsearch.hosts',
+  'monitoring.ui.elasticsearch',
+  'monitoring.xpack_api_polling_frequency_millis',
+  'server.uuid',
+  'server.name',
+  'server.host',
+  'server.port',
+  'monitoring.cluster_alerts.email_notifications.enabled',
+  'monitoring.cluster_alerts.email_notifications.email_address',
+  'monitoring.ui.ccs.enabled',
+  'monitoring.ui.elasticsearch.logFetchCount',
+];
+
+/**
+ * Invokes plugin modules to instantiate the Monitoring plugin for Kibana
+ * @param kibana {Object} Kibana plugin instance
+ * @return {Object} Monitoring UI Kibana plugin object
+ */
+export const monitoring = (kibana: LegacyPluginApi): LegacyPluginSpec => {
+  return new kibana.Plugin({
+    require: deps,
+    id: 'monitoring',
+    configPrefix: 'monitoring',
+    publicDir: resolve(__dirname, 'public'),
+    config: configDefaults,
+    uiExports: getUiExports(),
+    deprecations,
+
+    init(server: Server) {
+      const serverConfig = server.config();
+      const { getOSInfo, plugins, injectUiAppVars } = server as typeof server & { getOSInfo?: any };
+      const log = (...args: Parameters<typeof server.log>) => server.log(...args);
+      const route = (...args: Parameters<typeof server.route>) => server.route(...args);
+      const expose = (...args: Parameters<typeof server.expose>) => server.expose(...args);
+      const serverFacade = {
+        config: () => ({
+          get: (key: string) => {
+            if (validConfigOptions.includes(key)) {
+              return serverConfig.get(key);
+            }
+            throw new Error(`Unknown key '${key}'`);
+          },
+        }),
+        injectUiAppVars,
+        log,
+        logger: server.newPlatform.coreContext.logger,
+        getOSInfo,
+        events: {
+          on: (...args: Parameters<typeof server.events.on>) => server.events.on(...args),
+        },
+        route,
+        expose,
+        _hapi: server,
+        // @ts-ignore
+        _kbnServer: this.kbnServer as KbnServer,
+      };
+
+      const legacyPlugins = plugins as Partial<typeof plugins> & { infra?: InfraPlugin };
+      const { xpack_main, elasticsearch, infra, alerting } = legacyPlugins;
+      const {
+        core: coreSetup,
+        plugins: { usageCollection, licensing },
+      } = server.newPlatform.setup;
+
+      const pluginsSetup: PluginsSetup = {
+        usageCollection,
+        licensing,
+      };
+
+      const __LEGACY: LegacySetup = {
+        ...serverFacade,
+        plugins: {
+          xpack_main,
+          elasticsearch,
+          infra,
+          alerting,
+        },
+      };
+
+      new Plugin().setup(coreSetup, pluginsSetup, __LEGACY);
+    },
+
+    postInit(server: Server) {
+      const { infra } = server.plugins as Partial<typeof server.plugins> & { infra?: InfraPlugin };
```

Comment:
shouldn't you declare dependency on `infra` plugin?

Reply:
Do you mean through `kibana.json`? We'll do that in the next phase when we move the client outside the `legacy` directory

Reply:
not only, you declare dependencies on other plugins https://github.com/elastic/kibana/pull/51823/files#diff-3450f9bd06407cda6d6d9e26ddeb0e11R31-R35